benchmarks/src/main/scala/zio/schema/codec/ProtobufBenchmarks.scala

```diff
@@ -0,0 +1,46 @@
+package zio.schema.codec
+
+import org.openjdk.jmh.annotations._
+import zio.internal.{Platform, Tracing}
+import zio.{BootstrapRuntime, Chunk}
+import zio.schema.{DeriveSchema, Schema}
+import zio.stream.{ZSink, ZStream}
+
+import java.util.concurrent.TimeUnit
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Warmup(iterations = 10, time = 3, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 10, time = 3, timeUnit = TimeUnit.SECONDS)
+@Fork(1)
+@Threads(1)
+class ProtobufBenchmarks {
+  import CodecBenchmarks._
+
+  val TracedRuntime: BootstrapRuntime = new BootstrapRuntime {
+    override val platform = Platform.benchmark.withTracing(Tracing.enabled)
+  }
+
+  @Param(Array("1000"))
+  var size: Int = _
+
+  @Benchmark
+  def enumEncoding(): Chunk[Byte] = {
+    val output = Chunk.empty
+    for (i <- 0 until size) {
+      output ++ statusProtobufEncoder(statuses(i % 3))
+    }
+    output
+  }
+
+  @Benchmark
+  def enumDecoding(): Array[Status] = {
+    val output = Array.ofDim[Status](size)
```

Comment:
Best to push array allocation, etc., into `@setup` so it doesn't affect benchmark (actually this will be quite fast but just in general, a good idea to do setup outside the benchmark proper).

Reply:
Good call. Done.